src/components/CurrentCityWeather/CurrentCityWeather.js

```diff
@@ -100,19 +111,59 @@ const CurrentCityWeather = () => {
     setCity(addressComponents[0]);
   }
 
+
+  useEffect(() => {
+    if (transcript !== '') {
+      fetch(`https://maps.googleapis.com/maps/api/geocode/json?address=${transcript}&key=${process.env.REACT_APP_GOOGLE_MAPS_API_KEY}`)
+        .then(response => response.json())
+        .then(data => {
+          var latitude = data.results[0].geometry.location.lat;
+          var longitude = data.results[0].geometry.location.lng;
+          var coordinates = { lat: latitude, lon: longitude };
+          setAddress(data.results[0].formatted_address)
+          setCityCoordinates(coordinates);
+        })
+        .catch(error => alert(error))
+    }
+  }, [transcript])
+
+  function handleMicrophone() {
+    if (listening) {
+      SpeechRecognition.stopListening()
+    } else {
+      SpeechRecognition.startListening()
+    }
+  }
+
+  async function checkBraveBrowser() {
+    if (navigator.brave && await navigator.brave.isBrave()) {
+      return false
+    } else {
+      return true
+    }
+  }
+
   return (
     <>
       <div className="CurrentCityWeather">
         <h2 className="pb-4">Enter a city below 👇</h2>
-        <Autocomplete
-          apiKey={process.env.REACT_APP_GOOGLE_MAPS_API_KEY}
-          onPlaceSelected={(place) => {
-            setCoordinates(place);
-            getCity(place.formatted_address);
-          }}
-          defaultValue={"New York, NY, USA"}
-          className="inputCity"
-        />
+        <div className='input-ctn'>
+          <Autocomplete
+            apiKey={process.env.REACT_APP_GOOGLE_MAPS_API_KEY}
+            onPlaceSelected={(place) => {
+              setCoordinates(place);
+              getCity(place.formatted_address);
+            }}
+            defaultValue={"New York, NY, USA"}
+            className="inputCity"
+          />
+          {browserSupportsSpeechRecognition && checkBraveBrowser ? (
```

Comment:
You'll probably want to use `!` to make sure it's not Brave. Right now this code is making sure it is Brave.

Also, you'll have to be careful, `checkBraveBrowser` returns a `Promise` and a `Promise` is always true. However, you want to wait for the value returned by the promise before proceeding.

Reply:
checkBraveBrowser is actually a function I have defined that will return a boolean value.

Reply:
Actually, `checkBraveBrowser` will return the `Promise` of a `Boolean` since you used the async/await.

![image](https://user-images.githubusercontent.com/10343470/143816654-032d3d06-4030-4a1f-8ec5-124099f20645.png)

As you can see, it does if fact return a Promise and since a Promise is an Object, it's "true". Thus the function will always return "true" if it's not awaited or `.then`'ed.